engine/src/main/scala/cromwell/webservice/PartialWorkflowSources.scala

```diff
@@ -96,30 +100,51 @@ object PartialWorkflowSources {
       case _ => s"Incomplete workflow submission: $partialSource".invalidNel
     }
 
+    def validateWorkflowType(partialSource: PartialWorkflowSources): ErrorOr[Option[WorkflowType]] = {
+      partialSource.workflowType match {
+        case Some(src) => Option(src).validNel
```

Comment:
- What's the value of the `ErrorOr[]` wrapper if they're both valid?
- ToL: `Some(src) => Option(src)` makes me cry a little inside. Could it be:
 `Some(_) => partialSource.workflowType.validNel`
- ToL: it feels like there's probably some helper function somewhere along the lines of:
`partialSource.workflowType.noneMap(WorkflowOptions.defaultWorkflowType)`... 🤷‍♂️

Reply:
> What's the value of the ErrorOr[] wrapper if they're both valid?

So our wonderful cartesian ~~combine~~ tie-fighter operator may be used.

> ToL: Some(src) => Option(src) makes me cry a little inside.

Will do.

> ToL: it feels like there's probably some helper function somewhere along the lines of:
partialSource.workflowType.noneMap(WorkflowOptions.defaultWorkflowType)...

Probably is? I'm still absorbing cats knowledge from others. If the correct type-class appears before I merge, I'll use it! Otherwise, we can update these later.